### PR TITLE
Display original filename in editor

### PR DESCRIPTION
**Pull request by anediaz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

When importing a file, editor does not display the name of this file.
This can be confusing for users, who can forget the exact file they are working with (for example, if they use to work with multiple SketchApp designs).
Issue Number: #369 

## What is the new behavior?
The imported file's name appears on the header of the editor, between application's name (xLayers) and the action button toolbar

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/xlayers/src/app/editor/editor.component.html [hunk 1]
```diff
@@ -6,6 +6,10 @@
         >xLayers<span>{{ badge }}</span>
       </a>
     </div>
+    <div class="header-file-data">
+      <img *ngIf="fileData.logo" src={{fileData.logo}}/>
```
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/editor.component.html:10]:
```suggestion
      <img *ngIf="fileData.logo" [src]="fileData.logo"/>
```

File: libs/sketchtypes/src/lib/sketchtypes.ts [hunk 2]
```diff
@@ -483,4 +489,5 @@ export interface SketchMSData {
   document: SketchMSDocument;
   user: SketchMSUser;
   meta: SketchMSMeta;
+  fileData: SketchFile;
```
Comment (Jefiozie) [inline libs/sketchtypes/src/lib/sketchtypes.ts:492]: I'm not sure if this change is the correct one. The types we have defined here are types related to a Sketch Design file. Basically, this is our map of data of the ingested Sketch file.

I think we should remove it here and add this to the `UISettings` interface in our `ui.state.ts` as I believe this fits better in the generic parts of ingesting design files. 

Do you agree with this and what is your oppinion @manekinekko ? 

I'm happy to assist you with this btw, so DM me if you would like a chat.

**Review by Jefiozie — CHANGES_REQUESTED** [review]
I think we are almost there, I really like this change where you see the filename! 🚀

Could it be that the test around `editor.component` is broken? For some reason my machine is not running the tests so I cannot easly test this locally, could you verify this by running `ng run xlayers:test`?

File: libs/sketchtypes/src/lib/sketchtypes.ts (hunk 2 above)
Comment (anediaz) [reply libs/sketchtypes/src/lib/sketchtypes.ts:492]: Yes you're right. I added this quickly and I didn't think about the best place to put it.
Firstly I wanted to put it inside 'meta' object, but I realized it wasn't the right place either. I'll fix it.
Thank you for you help :-)

Comment (manekinekko) [reply libs/sketchtypes/src/lib/sketchtypes.ts:492]: Great work so far ☺️

I'd also suggest the following changes:
- change name from `SketchFile` to something generic like `DesignFile`
- change typing to:
```typescript
interface DesignFile {
  name: string;
  icon: string;
}
```
- And as @Jefiozie  suggested, move the file data description to the `UISettings`

File: apps/xlayers/src/app/supported-files.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Object that contains information about file types xlayers is compatible with
+ */
+const compatibleFiles = [
+  {
+    name: 'Sketch App',
+    extension: '.sketch',
+    logo: '/assets/supported/sketch.svg',
+    supportedVersions: [49, 50, 51, 52, 53, 59],
```
Comment (manekinekko) [inline apps/xlayers/src/app/supported-files.ts:9]: The `supportedVersions` can be computed from the environment config (see [environment.ts](https://github.com/xlayers/xlayers/blob/main/apps/xlayers/src/environments/environment.ts)).

File: apps/xlayers/src/app/supported-files.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Object that contains information about file types xlayers is compatible with
+ */
+const compatibleFiles = [
+  {
+    name: 'Sketch App',
+    extension: '.sketch',
+    logo: '/assets/supported/sketch.svg',
```
Comment (manekinekko) [inline apps/xlayers/src/app/supported-files.ts:8]:
```suggestion
    icon: '/assets/supported/sketch.svg',
```

File: apps/xlayers/src/app/supported-files.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * @param fileName the file's complete name (extension included)
+ */
+export function getFileData(fileName: string) {
+  const fileExtension = fileName.includes('.')
+    ? `.${fileName.split('.').pop()}`
+    : ''; // get file's extension with a point at the begining
+  const name = fileName.replace(fileExtension, ''); // extract file's name without the extension
+  const compatible = compatibleFiles.find(
+    ({ extension }) => fileExtension === extension
+  ) || { logo: '' }; // get information related to the file based on the extension
```
Comment (manekinekko) [inline apps/xlayers/src/app/supported-files.ts:24]:
```suggestion
  ) || { icon: '' }; // get information related to the file based on the extension
```

File: apps/xlayers/src/app/supported-files.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  return { name, logo: compatible.logo, extension: fileExtension }; //for the moment, minimum data is returned
```
Comment (manekinekko) [inline apps/xlayers/src/app/supported-files.ts:25]:
```suggestion
  return { name, icon: compatible.icon, extension: fileExtension }; //for the moment, minimum data is returned
```

**Comment by anediaz** [comment]
I've taken into account all your comments.
Could you please look at `ui.state.ts` file? I'm not sure my `sourceFileData`'s implementation is the best one.
I've also added a Cypress test to check if file's data is present on the editor as you suggested @Jefiozie

File: apps/xlayers/src/app/editor/editor.component.html [hunk 7]
```diff
@@ -5,6 +5,12 @@
       <a routerLink="/"
         >xLayers<span>{{ badge }}</span>
       </a>
+    </div>
+     <div  data-cy="fileData" class="header-file-data">
+      <ng-container *ngIf="sourceFileData as _data" >
+        <img [src]="_data.icon" />
+        <div class="file-name">{{_data.name}}</div>
+      </ng-container>
```
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/editor.component.html:13]:
```suggestion
     <div  data-cy="fileData" class="header-file-data">
      <ng-container *ngIf="sourceFileData as fileData" >
        <img [src]="fileData.icon" />
        <div class="file-name">{{fileData.name}}</div>
      </ng-container>
```

**Review by Jefiozie — COMMENTED** [review]
Just a small naming thing, further I think it looks good.

**Review by Jefiozie — APPROVED** [review]
LGTM

**Comment by anediaz** [comment]
> LGTM

Can I merge it ? 🤩

**Review by manekinekko — APPROVED** [review]

**Pull request merged**